templates/experiments/single_experiment_home.html

```diff
@@ -389,4 +504,139 @@ <h3 class="font-bold text-lg">Parent Experiments:
+
+            if (query === '') {
+              filter.filteredOptions = [...this.filterData.columns[column].options];
+            } else {
+              filter.filteredOptions = this.filterData.columns[column].options.filter(option =>
+                option.toLowerCase().includes(query)
+              );
+            }
+          }
+        },
+        removeFilter(index) {
+          this.filterData.filters.splice(index, 1);
+          this.triggerFilterChange();
+        },
+        updateFilterValue(index) {
+          const filter = this.filterData.filters[index];
+          if (['any of', 'all of'].includes(filter.operator)) {
+            filter.value = JSON.stringify(filter.selectedValues);
+          }
+          this.triggerFilterChange();
+        },
+        getActiveFilterCount() {
+          return this.filterData.filters.filter(filter => {
+            return filter.column && filter.operator && (
+              filter.value ||
+              (Array.isArray(filter.selectedValues) && filter.selectedValues.length > 0)
+            );
+          }).length;
+        },
+        clearAllFilters() {
+          this.filterData.filters = [];
+          this.addFilter();
+          this.filterData.showFilters = false;
+          this.triggerFilterChange();
+        },
+        triggerFilterChange() {
+          let filterParams = {};
+          this.filterData.filters.forEach((filter, i) => {
+            if (filter.column && filter.operator && (filter.value || filter.selectedValues.length > 0)) {
+              filterParams[`filter_${i}_column`] = filter.column;
+              filterParams[`filter_${i}_operator`] = filter.operator;
+              if (['any of', 'all of'].includes(filter.operator)) {
+                filterParams[`filter_${i}_value`] = JSON.stringify(filter.selectedValues);
+              } else {
+                filterParams[`filter_${i}_value`] = filter.value;
+              }
+            }
+          });
+          const url = new URL(window.location.href);
+          for (const key of [...url.searchParams.keys()]) {
+            if (key.startsWith('filter_')) {
+              url.searchParams.delete(key);
+            }
+          }
+          Object.entries(filterParams).forEach(([key, value]) => {
+            url.searchParams.set(key, value);
+          });
+          window.history.replaceState({}, '', url.toString());
+          const sessionsTableUrl = document.getElementById('sessions-table').getAttribute('hx-get');
+          const filterUrl = new URL(sessionsTableUrl, window.location.origin);
+          Object.entries(filterParams).forEach(([key, value]) => {
+            filterUrl.searchParams.set(key, value);
+          });
+          const showAllCheckbox = document.querySelector('input[name="show-all"]');
+          if (showAllCheckbox && showAllCheckbox.checked) {
+            filterUrl.searchParams.set('show-all', 'on');
+          }
+          htmx.ajax('GET', filterUrl.toString(), {
```

Comment:
Rather than generating the URL it might be easier to pass the data in the `values` field of the context object:

`htmx.ajax('GET', sessionsTableUrl, {..., values: filterParams})`

Reply:
much cleaner([a71e63e](https://github.com/dimagi/open-chat-studio/pull/1260/commits/a71e63e057a425126c006ceef0e7e9d0d752ca3d))